src/libutil/fmt.hh

```diff
@@ -63,63 +73,107 @@ inline std::string fmt(const std::string & fs, const Args & ... args)
     return f.str();
 }
 
-// -----------------------------------------------------------------------------
-// format function for hints in errors.  same as fmt, except templated values
-// are always in yellow.
-
+/**
+ * Values wrapped in this struct are printed in magenta.
+ *
+ * By default, arguments to `HintFmt` are printed in magenta. To avoid this,
+ * either wrap the argument in `Uncolored` or add a specialization of
+ * `HintFmt::operator%`.
+ */
 template <class T>
-struct yellowtxt
+struct Magenta
 {
-    yellowtxt(const T &s) : value(s) {}
+    Magenta(const T &s) : value(s) {}
     const T & value;
 };
 
 template <class T>
-std::ostream & operator<<(std::ostream & out, const yellowtxt<T> & y)
+std::ostream & operator<<(std::ostream & out, const Magenta<T> & y)
 {
     return out << ANSI_WARNING << y.value << ANSI_NORMAL;
 }
 
+/**
+ * Values wrapped in this class are printed without coloring.
+ *
+ * By default, arguments to `HintFmt` are printed in magenta (see `Magenta`).
+ */
 template <class T>
-struct normaltxt
+struct Uncolored
 {
-    normaltxt(const T & s) : value(s) {}
+    Uncolored(const T & s) : value(s) {}
     const T & value;
 };
 
 template <class T>
-std::ostream & operator<<(std::ostream & out, const normaltxt<T> & y)
+std::ostream & operator<<(std::ostream & out, const Uncolored<T> & y)
 {
     return out << ANSI_NORMAL << y.value;
 }
 
-class hintformat
+/**
+ * A wrapper around `boost::format` which colors interpolated arguments in
+ * magenta by default.
+ */
+class HintFmt
 {
+private:
+    boost::format fmt;
+
 public:
-    hintformat(const std::string & format) : fmt(format)
+    /**
+     * Construct a `HintFmt` from a format string, with values to be
+     * interpolated later with `%`.
+     *
+     * This isn't exposed as a single-argument constructor to avoid
+     * accidentally constructing `HintFmt`s with user-controlled strings. See
+     * the note on `fmt` for more information.
+     */
+    static HintFmt interpolate(const std::string & formatString)
+    {
+        HintFmt result((boost::format(formatString)));
+        result.fmt.exceptions(
+            boost::io::all_error_bits ^
+            boost::io::too_many_args_bit ^
+            boost::io::too_few_args_bit);
+        return result;
+    }
```

Comment:
This doesn't seem to be called anywhere.

I'm a bit unclear as to the semantics of everything, but doesn't that mean that we never apply the `.exceptions` call (and end-up with whatever is the default, which isn't totally clear from the [boost::format docs](https://www.boost.org/doc/libs/1_84_0/libs/format/doc/format.html).

Reply:
Oh, good catch! I've removed this function and added a `setExceptions` function to call in the appropriate constructor.